### PR TITLE
only regenerate scheduled from today

### PR DESCRIPTION
**Pull request by paul-watkins-croud**

### CODE REVIEW

File: src/Repository/ScheduleRepository.php [hunk 1]
```diff
@@ -57,16 +60,20 @@ public function updateFromScheduleDefinition(int $schedule_id, array $definition
      * @return Collection
      */   
 
-    public function regenerateScheduleEvents($schedule) : Collection
+        public function regenerateScheduleEvents($schedule) : Collection
```
Comment (matthewerskine) [inline src/Repository/ScheduleRepository.php]: Unnecessary formatting

**Review by matthewerskine — APPROVED** [review]

File: src/Repository/ScheduleRepository.php [hunk 2]
```diff
@@ -71,8 +71,8 @@ public function regenerateScheduleEvents($schedule) : Collection
         $parser_dates = $parser->getDates();
         $current_dates = $schedule->scheduleEvents()->get()->pluck('date');
         $deprecated_dates = collect($current_dates)->diff($parser_dates)->toArray();
-        $new_dates = collect($parser_dates)->diff($current_dates)->filter(function($date) use ($today) {
-            return $date >= $today;
+        $new_dates = collect($parser_dates)->diff($current_dates)->filter(function($date) use ($today, $from_today) {
```
Comment (matthewerskine) [inline src/Repository/ScheduleRepository.php]: If you need an external conditional (e.g. `$from_today`) you can also write this using `when` as shown below:

```php
$new_dates = collect($parser_dates)->diff($current_dates)->when($from_today, function($date) use ($today) {
    return $date >= $today;
});
```

Comment (paul-watkins-croud) [reply src/Repository/ScheduleRepository.php]: Nice. I'll tweak that

Comment (paul-watkins-croud) [reply src/Repository/ScheduleRepository.php]: Ah, not on 5.3 though

Comment (matthewerskine) [reply src/Repository/ScheduleRepository.php]: My bad.
